SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,254 @@
+  CTurbomachineryPrimitiveState OutletPrimitiveState;
+  public:
+  CTurbomachineryCombinedPrimitiveStates();
+
+  CTurbomachineryCombinedPrimitiveStates(const CTurbomachineryPrimitiveState &inletPrimitiveState,
+                                         const CTurbomachineryPrimitiveState &outletPrimitiveState);
+
+  ~CTurbomachineryCombinedPrimitiveStates();
+
+  CTurbomachineryPrimitiveState GetInletPrimitiveState() const { return InletPrimitiveState; }
+
+  CTurbomachineryPrimitiveState GetOutletPrimitiveState() const { return OutletPrimitiveState; }
+};
+
+class CTurbomachineryState {
+  private:
+  su2double Density, Pressure, Entropy, Enthalpy, Temperature, TotalTemperature, TotalPressure, TotalEnthalpy;
+  su2double AbsFlowAngle, FlowAngle, MassFlow, Rothalpy, TotalRelPressure;
+  vector <su2double> Velocity, RelVelocity, Mach, RelMach;
+  su2double Area, Radius;
+
+
+  public:
+  CTurbomachineryState();
+
+  CTurbomachineryState(unsigned short nDim, su2double area, su2double radius);
+
+  ~CTurbomachineryState();
+
+  void ComputeState(CFluidModel *const fluidModel, const CTurbomachineryPrimitiveState &primitiveState);
+
+  su2double GetDensity() const { return Density; }
+
+  su2double GetPressure() const { return Pressure; }
+
+  su2double GetEntropy() const { return Entropy; }
+
+  su2double GetEnthalpy() const { return Enthalpy; }
+
+  su2double GetTemperature() const { return Temperature; }
+
+  su2double GetTotalTemerature() const { return TotalTemperature; }
+
+  su2double GetTotalPressure() const { return TotalPressure; }
+
+  su2double GetTotalRelPressure() const { return TotalRelPressure; }
+
+  su2double GetTotalEnthalpy() const { return TotalEnthalpy; }
+
+  su2double GetAbsFlowAngle() const { return AbsFlowAngle; }
+
+  su2double GetFlowAngle() const { return FlowAngle; }
+
+  su2double GetMassFlow() const { return MassFlow; }
+
+  su2double GetRothalpy() const { return Rothalpy; }
+
+  vector <su2double> GetVelocity() const { return Velocity; }
+
+  vector <su2double> GetMach() const { return Mach; }
+
+  su2double GetVelocityValue() const {
+    return sqrt(inner_product(Velocity.begin(), Velocity.end(), Velocity.begin(), 0.0));
+  }
+
+  su2double GetMachValue() const { return sqrt(inner_product(Mach.begin(), Mach.end(), Mach.begin(), 0.0)); }
+
+  su2double GetRelVelocityValue() const {
+    return sqrt(inner_product(RelVelocity.begin(), RelVelocity.end(), RelVelocity.begin(), 0.0));
+  }
+
+  su2double GetRelMachValue() const {
+    return sqrt(inner_product(RelMach.begin(), RelMach.end(), RelMach.begin(), 0.0));
+  }
+};
+
+
+class CTurbomachineryBladePerformance {
+  protected:
+  CTurbomachineryState InletState;
+  CTurbomachineryState OutletState;
+  su2double KineticEnergyLoss, TotalPressureLoss, EntropyGen, PressureRatio, EulerianWork;
+  CFluidModel *FluidModel;
+
+  public:
+  CTurbomachineryBladePerformance(CFluidModel *const fluidModel, unsigned short nDim, su2double areaIn,
```

Comment:
This usage of `*const` (and in other methods too) is somewhat misleading, as it means the pointer is const but the object pointed to is not. Also, references should be preferred IMO as they live no room for passing nullptr.